### PR TITLE
[projects] clear team selection

### PR DESCRIPTION
**Pull request by AlexTugarev**
Followup to https://github.com/gitpod-io/gitpod/pull/5854

In this change we clear the team selection whenever a non-team page is loaded.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **alextugarev** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Replying to https://github.com/gitpod-io/gitpod/pull/5854#issuecomment-928899776 in this open PR:

> ... but I think unsetting when navigating away from a team's page sounds reasonable.

Agree and this already happens for example when you visit the admin dashboard, right?

One edge case is what happens when a user is removed from a team or a team is deleted. Probably defaulting back to the personal account sounds reasonable, right? 🧀  

See also [relevant discussion](https://gitpod.slack.com/archives/C02EN94AEPL/p1632490746030200) (internal).

Cc @jankeromnes @AlexTugarev

**Review by jankeromnes — APPROVED** [review]
Works as advertised! Many thanks for the quick fix.

I still think it's dangerous & only slightly helpful to add this kind of "magic behavior" to our app, but I'm okay with this for now. Maybe let's remove this feature if we notice that it creates new bugs? (We already have a lot of [feature creep](https://en.wikipedia.org/wiki/Feature_creep) in Gitpod).

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: db4f4da203aa9aa776d8473ad0d861371ae9848a</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5894#pullrequestreview-765100094" title="Approved">jankeromnes</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5894#issuecomment-928985354" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5894#pullrequestreview-765100094" title="Approved">jankeromnes</a>*

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5894#issuecomment-928985354" title="Approved">AlexTugarev</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev,jankeromnes]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
